pandas/_libs/tslibs/conversion.pyx

```diff
@@ -166,34 +163,21 @@ cpdef inline (int64_t, int) precision_from_unit(
         int p
         NPY_DATETIMEUNIT reso = abbrev_to_npy_unit(unit)
 
-    multiplier = periods_per_second(out_reso)
-
+    if reso == NPY_DATETIMEUNIT.NPY_FR_GENERIC:
+        reso = NPY_DATETIMEUNIT.NPY_FR_ns
     if reso == NPY_DATETIMEUNIT.NPY_FR_Y:
         # each 400 years we have 97 leap years, for an average of 97/400=.2425
         #  extra days each year. We get 31556952 by writing
         #  3600*24*365.2425=31556952
+        multiplier = periods_per_second(out_reso)
         m = multiplier * 31556952
     elif reso == NPY_DATETIMEUNIT.NPY_FR_M:
         # 2629746 comes from dividing the "Y" case by 12.
+        multiplier = periods_per_second(out_reso)
         m = multiplier * 2629746
-    elif reso == NPY_DATETIMEUNIT.NPY_FR_W:
-        m = multiplier * 3600 * 24 * 7
-    elif reso == NPY_DATETIMEUNIT.NPY_FR_D:
-        m = multiplier * 3600 * 24
-    elif reso == NPY_DATETIMEUNIT.NPY_FR_h:
-        m = multiplier * 3600
-    elif reso == NPY_DATETIMEUNIT.NPY_FR_m:
-        m = multiplier * 60
-    elif reso == NPY_DATETIMEUNIT.NPY_FR_s:
-        m = multiplier
-    elif reso == NPY_DATETIMEUNIT.NPY_FR_ms:
-        m = multiplier // 1_000
-    elif reso == NPY_DATETIMEUNIT.NPY_FR_us:
-        m = multiplier // 1_000_000
-    elif reso == NPY_DATETIMEUNIT.NPY_FR_ns or reso == NPY_DATETIMEUNIT.NPY_FR_GENERIC:
-        m = multiplier // 1_000_000_000
     else:
-        raise ValueError(f"cannot cast unit {unit}")
+        m = get_conversion_factor(reso, out_reso)
```

Comment:
if `get_conversion_factor` raises, then `precision_from_unit` wouldn't propagate the error - perhaps a note could be made about this in the docstring, alongside the current note?

Reply:
comment sounds good

Reply:
to keep from clogging the CI, mind if i add this comment in my next CLN: assorted branch?

Reply:
sounds good!